uamqp/message.py

```diff
@@ -727,21 +806,33 @@ def __init__(self,
             self.reply_to_group_id = reply_to_group_id
 
     def __str__(self):
-        return str({
-            'message_id': self.message_id,
-            'user_id': self.user_id,
-            'to': self.to,
-            'subject': self.subject,
-            'reply_to': self.reply_to,
-            'correlation_id': self.correlation_id,
-            'content_type': self.content_type,
-            'content_encoding': self.content_encoding,
-            'absolute_expiry_time': self.absolute_expiry_time,
-            'creation_time': self.creation_time,
-            'group_id': self.group_id,
-            'group_sequence': self.group_sequence,
-            'reply_to_group_id': self.reply_to_group_id
-        })
+        return str(
+            {
+                "message_id": self.message_id,
+                "user_id": self.user_id,
+                "to": self.to,
+                "subject": self.subject,
+                "reply_to": self.reply_to,
+                "correlation_id": self.correlation_id,
+                "content_type": self.content_type,
+                "content_encoding": self.content_encoding,
+                "absolute_expiry_time": self.absolute_expiry_time,
+                "creation_time": self.creation_time,
+                "group_id": self.group_id,
+                "group_sequence": self.group_sequence,
+                "reply_to_group_id": self.reply_to_group_id,
+            }
+        )
+
+    def __getstate__(self):
+        state = self._get_properties_dict()
+        state["_encoding"] = self._encoding
```

Comment:
I guess pickle should have its own encode format support? we probably don't need to include our encoding information in the state.

Reply:
This encoding is the attribute associated with MessageProperties. I'm including it here because, in `__setstate__`, I reset the encoding of the MessageProperties object because it is passed into the `utils.data_factory(value, encoding=self._encoding)` call when resetting all of the attributes. I thought this encoding was separate from the pickle encoding, but my understanding of encoding might be wrong?